MaterialLib/SolidModels/CreepBGRa.cpp

```diff
@@ -47,25 +47,25 @@ CreepBGRa<DisplacementDim>::integrateStress(
         std::exp(-_Q / (MaterialLib::PhysicalConstant::IdealGasConstant * T));
     auto const& deviatoric_matrix = Invariants::deviatoric_projection;
 
-    auto const update_residual = [&](ResidualVectorType& r) {
-
-        auto const s_n1 = deviatoric_matrix * solution;
-        // ||s_{n+1}|| = sqrt(2.0 * J2)
-        double const norm_s_n1 = std::sqrt(2.0 * Invariants::J2(s_n1));
-        r = solution - sigma_try +
-            2.0 * b * this->_mp.mu(t, x) * std::pow(norm_s_n1, _n - 1) * s_n1;
-    };
+    double pow_norm_s_n1_n_minus_one_2b_G = 0.;
 
+    KelvinVector s_n1;
     auto const update_jacobian = [&](JacobianMatrix& jacobian) {
-        auto const s_n1 = deviatoric_matrix * solution;
+        s_n1 = deviatoric_matrix * solution;
         double const norm_s_n1 = std::sqrt(2.0 * Invariants::J2(s_n1));
+        pow_norm_s_n1_n_minus_one_2b_G =
```

Comment:
Oh no, now `update_jacobian` is not a pure function anymore, but has side effects. :smile: 
Maybe there's a performance gain from that. But if there are errors or refactorings, the code is harder to understand.
I'd maybe add comments `// side effect` to the two assignments above.

Reply:
Yes, the only reason is the performance. I have added a little long comment above the definitions of the two variables to explain the reason.